codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/rendering/samples/KDocSamplesGenerator.kt

```diff
@@ -0,0 +1,143 @@
+
+    private fun sampleFunctionName(index: Int): String = "sample${index + 1}"
+    private fun sampleClassName(op: OperationShape): String = op.id.name
+
+    private fun samplePackage(settings: KotlinSettings): String = settings.pkg.subpackage("samples")
+    override fun writeAdditionalFiles(ctx: CodegenContext, delegator: KotlinDelegator) {
+        val topDownIndex = TopDownIndex.of(ctx.model)
+        val operations = topDownIndex.getContainedOperations(ctx.settings.service)
+        operations.filter { it.hasTrait<ExamplesTrait>() }
+            .forEach { op ->
+                val examples = op.expectTrait<ExamplesTrait>()
+
+                val writer = KotlinWriter(samplePackage(ctx.settings))
+                writer.withBlock("class #L {", "}", sampleClassName(op)) {
+                    examples.examples
+                        // unclear what benefit error examples provide, omit for now
+                        .filterNot { it.error.isPresent }
+                        .forEachIndexed { idx, example ->
+                            write("")
+                            write("@Sample")
+                            withBlock("fun #L() {", "}", sampleFunctionName(idx)) {
+                                example.documentation
+                                    .getOrDefault(example.title)
+                                    .breakLongLines()
+                                    .forEach { line ->
+                                        write("// #L", line)
+                                    }
+
+                                renderNormalExample(ctx, writer, op, example)
+                            }
+                            write("")
+                        }
+                }
+                val contents = writer.toString()
+                delegator.fileManifest.writeFile("src/samples/${op.id.name}.kt", contents)
+            }
+    }
+
+    private fun renderNormalExample(ctx: CodegenContext, writer: KotlinWriter, op: OperationShape, example: Example) {
+        val clientName = clientName(ctx.settings.sdkId).replaceFirstChar { it.lowercase(Locale.getDefault()) }
+        val respPrefix = if (example.output.isPresent) "val resp = " else ""
+
+        val input = ctx.model.expectShape(op.inputShape)
+        if (input.hasAllOptionalMembers && example.input.isEmpty) {
+            writer.write("#L#LClient.#L()", respPrefix, clientName, op.defaultName())
+        } else {
+            writer.withBlock("#L#LClient.#L {", "}", respPrefix, clientName, op.defaultName()) {
+                ShapeValueGenerator(ctx.model, ctx.symbolProvider).writeShapeValues(writer, input, example.input)
+            }
+        }
+
+        // TODO - not clear what benefit there is to the example output, omit it for now
+    }
+}
+
+internal fun String.breakLongLines(maxLineLengthChars: Int = 100): List<String> {
+    val words = Regex("\\w+[.,]?|\\\".*\\\"[.,]?|\\(.*\\)[.,]?").findAll(this).map(MatchResult::value)
```

Comment:
**Correctness**: The quoted string and parenthesized string cases use greedy matching, which will cause problems in strings with multiple quoted or parenthesized sections. For example, the pattern applied to `The "quick" brown fox jumped over the "lazy" dogs.` would match `"quick" brown fox jumped over the "lazy"`. I think you want to use `.*?` instead of `.*`.